Push Failure after Restoration of from Backup

Hi,
I just did a restored my apps from a backup as described [here](http://docs.deis.io/en/latest/managing_deis/backing_up_data/) without any major issues except [this which was not backing up](http://docs.deis.io/en/latest/managing_deis/backing_up_data/#database-data).

When i try to push new changes  to my app , the git push fails (only via the buildpacks workflow , there are no problems with the Dockerfile workflow), but from my little observation, the push seems to be pushing the entire code base and not the latest changes, I thought this might be a good starting point.
Please see [gist](https://gist.github.com/babarinde/b9def784d4cbc576fcc0#file-gistfile1-txt)
Also I just need to be sure that when doing a recovery, does the new  deis platform version have to be the same version as the backed up one or is not necessary?

Thanks

@bacongobbler 
After a while at the launching stage it shows this:
-----> Launching... 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0x40179b]

goroutine 1 [running]:
main.main()
remote: /home/jenkins/workspace/test-acceptance/src/github.com/deis/deis/builder/bin/publish-release-controller.go:80 +0xb5b
To ssh://git@deis.abacus.com.ng:2222/ischool.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'ssh://git@deis.abacus.com.ng:2222/ischool.git'

--- a/builder/image/slugbuilder/builder/install-buildpacks
+++ b/builder/image/slugbuilder/builder/install-buildpacks
@@ -24,7 +24,7 @@ download_buildpack https://github.com/heroku/heroku-buildpack-gradle.git        
 download_buildpack https://github.com/heroku/heroku-buildpack-grails.git         1ef927d
 download_buildpack https://github.com/heroku/heroku-buildpack-play.git           9c137b4
 download_buildpack https://github.com/heroku/heroku-buildpack-python.git         v53
-download_buildpack https://github.com/heroku/heroku-buildpack-php.git            v50
+download_buildpack https://github.com/heroku/heroku-buildpack-php.git            v57
 download_buildpack https://github.com/heroku/heroku-buildpack-clojure.git        v63
 download_buildpack https://github.com/heroku/heroku-buildpack-scala.git          v43
 download_buildpack https://github.com/heroku/heroku-buildpack-go.git             0e20030

